server/lib/nl/fulfillment/time_delta_across_places.py

```diff
@@ -127,15 +128,24 @@ def populate(state: PopulateState, chart_vars: ChartVars, places: List[Place],
     for d in ranked_dcids:
       ranked_places.append(dcid2place[d])
     ranked_places = ranked_places[:constants.MAX_ANSWER_PLACES]
+    place_facet_ids = {}
+    for place, facet in place_facets.items():
+      place_facet_ids[place] = facet.get('facetId', '')
 
     # TODO: Uncomment this once we agree on look and feel
     if field == 'abs' and ranked_places:
+      sv_place_latest_date = {}
+      _, end_date = get_date_range_strings(state.date_range)
+      if end_date:
+        sv_place_latest_date = utils.get_predicted_latest_date(
+            {sv: place_facets}, state.date_range)
       found |= add_chart_to_utterance(ChartType.TIMELINE_WITH_HIGHLIGHT,
```

Comment:
hmm, should this happen only when `state.date_range` is set?

Reply:
yes, if state.date_range is not set, end_date will be empty